### PR TITLE
fix issues 540 and 542 : Rename reverse argment as ascending in sorting methods + fixed bug in sort_values

### PR DESCRIPTION
**Pull request by alixdamman**

### CODE REVIEW

File: doc/source/changes/version_0_27.rst.inc [hunk 1]
```diff
@@ -171,3 +174,5 @@ Fixes
 -----
 
 * fixed array values being editable in view() (instead of only in edit()).
+
+* fixed `sort_values` failing when passing only `ascending` argument (closes :issue:`542`).
```
Comment (gdementen) [inline doc/source/changes/version_0_27.rst.inc]: remove from changelog as it was not present in 0.26.1 (could not call sort_values without key).

File: doc/source/changes/version_0_27.rst.inc [hunk 2]
```diff
@@ -3,6 +3,9 @@
 
 * renamed `Axis.translate` to `Axis.index` (closes :issue:`479`).
 
+* deprecated `reverse` argument of `sort_values`, `sort_axes`, `labelsofsorted` and `indicesofsorted` methods
```
Comment (gdementen) [inline doc/source/changes/version_0_27.rst.inc]: remove mention of `indicesofsorted` and `labelsofsorted` here (they did not support reverse in 0.26.1). Also need to update the changelog for those which still says implemented "reverse"

File: larray/core/array.py [hunk 3]
```diff
@@ -3134,7 +3135,8 @@ def indexofmax(self, axis=None):
 
     posargmax = renamed_to(indexofmax, 'posargmax')
 
-    def labelsofsorted(self, axis=None, reverse=False, kind='quicksort'):
+    @deprecate_kwarg('reverse', 'ascending', {True: False, False: True})
```
Comment (gdementen) [inline larray/core/array.py]: no need for the deprecation here

File: larray/core/array.py [hunk 4]
```diff
@@ -3179,12 +3179,13 @@ def labelsofsorted(self, axis=None, reverse=False, kind='quicksort'):
...
             axis = self.axes[0]
         axis = self.axes[axis]
-        pos = self.indicesofsorted(axis, reverse=reverse, kind=kind)
+        pos = self.indicesofsorted(axis, ascending=ascending, kind=kind)
         return LArray(axis.labels[pos.data], pos.axes)
 
     argsort = renamed_to(labelsofsorted, 'argsort')
 
-    def indicesofsorted(self, axis=None, reverse=False, kind='quicksort'):
+    @deprecate_kwarg('reverse', 'ascending', {True: False, False: True})
```
Comment (gdementen) [inline larray/core/array.py]: no need for the deprecation here

File: larray/util/misc.py [hunk 5]
```diff
@@ -627,6 +627,36 @@ def wrapper(*args, **kwargs):
...
+                else:
+                    new_arg_value = old_arg_value
+                    msg = "The '{old_name}' keyword is deprecated, use '{new_name}' instead"\
+                        .format(old_name=old_arg_name, new_name=new_arg_name)
+
+                warnings.warn(msg, FutureWarning, stacklevel=stacklevel)
+                if kwargs.get(new_arg_name, None) is not None:
+                    msg = "Can only specify '{old_name}' or '{new_name}', not both"\
+                        .format(old_name=old_arg_name, new_name=new_arg_name)
+                    raise TypeError(msg)
```
Comment (gdementen) [inline larray/util/misc.py]: should be ValueError, not TypeError

File: larray/util/misc.py [hunk 6]
```diff
@@ -627,6 +627,36 @@ def wrapper(*args, **kwargs):
...
+                    msg = "The {old_name}={old_val!r} keyword is deprecated, use {new_name}={new_val!r} instead"\
+                        .format(old_name=old_arg_name, old_val=old_arg_value, new_name=new_arg_name,
+                                new_val=new_arg_value)
+                else:
+                    new_arg_value = old_arg_value
+                    msg = "The '{old_name}' keyword is deprecated, use '{new_name}' instead"\
+                        .format(old_name=old_arg_name, new_name=new_arg_name)
+
+                warnings.warn(msg, FutureWarning, stacklevel=stacklevel)
+                if kwargs.get(new_arg_name, None) is not None:
```
Comment (gdementen) [inline larray/util/misc.py]: `if new_arg_name in kwargs:` ?

Comment (alixdamman) [reply larray/util/misc.py]: This decorator comes directly from Pandas and may be a little old...

Comment (gdementen) [reply larray/util/misc.py]: tsss... We cannot just copy-paste code from Pandas without any notice...

**Pull request merged**